src/Umbraco.Web.UI.Client/src/views/media/media.restore.controller.js

```diff
@@ -0,0 +1,72 @@
+angular.module("umbraco").controller("Umbraco.Editors.Media.RestoreController",
+	function ($scope, relationResource, mediaResource, navigationService, appState, treeService) {
+		var dialogOptions = $scope.dialogOptions;
+
+		var node = dialogOptions.currentNode;
+
+		$scope.error = null;
+	    $scope.success = false;
+
+		relationResource.getByChildId(node.id, "relateParentDocumentOnDelete").then(function (data) {
+
+            if (data.length == 0) {
+                $scope.success = false;
+                $scope.error = {
+                    errorMsg: "Cannot automatically restore this item",
+                    data: {
+                        Message: "There is no 'restore' relation found for this node. Use the Move menu item to move it manually."
```

Comment:
These texts should be moved to a language file and use the angular localization service to get them